tests/test_queue.py

```diff
@@ -42,241 +37,176 @@ def items():
 
 
 @pytest.mark.gen_test
-def test_unbounded_put_then_get(items):
+def test_put_then_get(items):
     queue = Queue()
-    assert_invariant(queue)
 
     for item in items:
         yield queue.put(item)
-        assert_invariant(queue)
 
     got_futures = []
     for i in range(len(items)):
         got_futures.append(queue.get())
-        assert_invariant(queue)
 
     got = yield got_futures
     assert got == items
 
-    with pytest.raises(TimeoutError):
-        yield queue.get(timedelta(seconds=0.1))
-
-
-@pytest.mark.gen_test
-def test_bounded_put_then_get(items):
-    queue = Queue(5)
-    assert_invariant(queue)
-
-    for item in items[:5]:
-        yield queue.put(item)
-        assert_invariant(queue)
-
-    put_futures = []
-    for item in items[5:]:
-        put_futures.append(queue.put(item))
-
-    got = []
-    for i in range(len(items)):
-        item = yield queue.get()
-        assert_invariant(queue)
-        got.append(item)
-
-    assert got == items
-    yield put_futures
-
 
 @pytest.mark.gen_test
-def test_bounded_put_timeout_then_get(items):
-    queue = Queue(1)
-    assert_invariant(queue)
-
-    yield queue.put(items[0])
-
-    put_futures = []
-    for item in range(len(items)):
-        put_futures.append(queue.put(item, timedelta(seconds=0.001)))
-
-    yield gen.sleep(0.01)
-
-    get_futures = []
-    for item in range(len(items)):
-        get_futures.append(queue.get())
-
-    for future in put_futures:
-        with pytest.raises(TimeoutError):
-            yield future
-
-    for item in items[1:]:
-        yield queue.put(item)
-
-    got = yield get_futures
-    assert got == items
-
-
-@pytest.mark.gen_test
-def test_unbounded_put_then_get_nowait(items):
+def test_put_then_get_nowait(items):
     queue = Queue()
 
     for item in items:
         yield queue.put(item)
-        assert_invariant(queue)
 
     got = []
     for i in range(len(items)):
         got.append(queue.get_nowait())
-        assert_invariant(queue)
 
     assert got == items
 
     with pytest.raises(QueueEmpty):
         queue.get_nowait()
 
+    future = queue.get()
+    yield queue.put(42)
+    assert 42 == (yield future)
+
 
 @pytest.mark.gen_test
-def test_unbounded_get_then_put(items):
+def test_get_then_put(items):
     queue = Queue()
 
     got_futures = []
     for i in range(len(items)):
         got_futures.append(queue.get())
-        assert_invariant(queue)
 
     for item in items:
         yield queue.put(item)
-        assert_invariant(queue)
 
     got = yield got_futures
     assert got == items
 
 
 @pytest.mark.gen_test
-def test_unbounded_expired_gets_then_get_first():
-    queue = Queue()
+@pytest.mark.concurrency_test
```

Comment:
Not a lot. It takes a couple seconds. I wanted to be able to skip that when I was verifying queue functionality.